### ISSUE TITLE
NSFW/NonNsfw based display filtering 

### ISSUE THREAD

**Issue by awcator** [tachiyomiorg/tachiyomi#10049]
### Describe your suggested feature

I'd like to see a feature in Tachiyomi, inspired by the Reddit app, where I can control the visibility of NSFW (Not Safe for Work) content in my feeds. Similarly, With a single user account or profile, I should have the option to enable or disable NSFW content, and when it's turned off, such content should not appear in my library. So that I can open the app even in public and start reading mangas. 

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have updated the app to version **[0.14.6](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

**Comment by CursedSpiderBoi** [comment]
less do this

**Comment by GZMaster** [comment]
Another idea would be to archive selected tags if mangas in the library

**Comment by SatoX69** [comment]
Would be great if they added one for categories too

**Comment by CursedSpiderBoi** [comment]
the idea i have in mind is that give user a options of blacklisting certain tags, and when displaying list from sources, cross check from the blacklisted tags and dont show them

**Comment by priyanshudebnath** [comment]
@CursedSpiderBoi you are right. This will be much better than doing this only for NSFW/SFW. 
Ability to make tags and hide any tag.

**Comment by CursedSpiderBoi** [comment]
I shall start work on this then should take not long if my system support me 😭

**Comment by arkon** [comment]
> the idea i have in mind is that give user a options of blacklisting certain tags, and when displaying list from sources, cross check from the blacklisted tags and dont show them

You can't do this unless you load the details of every entry, which we wouldn't allow given it'd massively increase the amount of traffic to sources. You'd be able to do it in the library, but you can already search by tags.

If you want NSFW/SFW toggling, it'd at best be broadly based on whether an extension/source is marked as NSFW or not.

**Comment by SatoX69** [comment]
How about this, You can set certain custom tags (not related to the
sources) if I hold a manga cover, it will prompt Add Tags and if I set the
tag of the manga in question to the blacklisted tag then check the filter
dialogue box, it won't show those mangas for as long as its toggled

**Comment by arkon** [comment]
#2371 would be far simpler than something like that.

**Comment by Lahnshen** [comment]
> > the idea i have in mind is that give user a options of blacklisting certain tags, and when displaying list from sources, cross check from the blacklisted tags and dont show them
> 
> You can't do this unless you load the details of every entry, which we wouldn't allow given it'd massively increase the amount of traffic to sources. You'd be able to do it in the library, but you can already search by tags.
> 
> If you want NSFW/SFW toggling, it'd at best be broadly based on whether an extension/source is marked as NSFW or not.

You're right about this but since that's the case then it might be good to give users ability to mark some titles as NSFW by themselves and it could then be filtered according to the filtering suggested by OP.
